### ISSUE TITLE
SCDF starts composed-task-runner instead  of my task after LDAP has enabled

### ISSUE THREAD

**Issue by Rostish** [spring-cloud/spring-cloud-dataflow#2452]
Good day collegues.
I have found very strange behavior.

I have enabled LDAP with next start conf.:

```
    #!/usr/bin/env bash
    export spring_datasource_url=jdbc:postgresql://xx.xxx.xx.xx:5432/data_flow
    export spring_datasource_username=data_flow_main
    export spring_datasource_password=xx
    export spring_datasource_driver_class_name=org.postgresql.Driver
    java -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005 \
    -Djavax.net.debug=ssl:handshake:verbose \
    -jar /mnt/store/viewing-maker/base-operations/scdf/spring-cloud-dataflow-server-local-1.7.0.BUILD- 
    SNAPSHOT.jar \
    --spring.cloud.dataflow.task.maximum-concurrent-tasks=300 \
    --security.basic.enabled=true \
    --spring.cloud.dataflow.security.authentication.ldap.enabled=true \
    --spring.cloud.dataflow.security.authentication.ldap.url="ldap://xxx.xx:389" \
    --spring.cloud.dataflow.security.authentication.ldap.managerDn="CN=xxxx,OU=Служебные 
   пользователи,DC=xxxx,DC=xxx" \
   --spring.cloud.dataflow.security.authentication.ldap.managerPassword="xxxxx" \
   --spring.cloud.dataflow.security.authentication.ldap.userSearchBase="dc=xxxxxx,dc=xxx" \
   --spring.cloud.dataflow.security.authentication.ldap.userSearchFilter="samaccountname={0}" \
   --spring.cloud.dataflow.security.authentication.ldap.groupSearchBase="OU=Глобальные 
   Группы,DC=xxxxx,DC=xxx" \
   --spring.cloud.dataflow.security.authentication.ldap.groupSearchFilter="member={0}" \
   --spring.cloud.dataflow.security.authentication.ldap.roleMappings.ROLE_MANAGE="BIGRODUCTION" \
   --spring.cloud.dataflow.security.authentication.ldap.roleMappings.ROLE_VIEW="BIGTRODUCTION" \
   --spring.cloud.dataflow.security.authentication.ldap.roleMappings.ROLE_CREATE="BIGTRODUCTION" \
```
Authorisation works. I tried to sent command via REST API:

```   
private HttpHeaders getHttpHeaders() {
        HttpHeaders headers = new HttpHeaders();
        String plainCredentials = String.format("%1$s:%2$s", userName, password);
        String base64Credentials = Base64.getEncoder().encodeToString(plainCredentials.getBytes());
        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
        headers.add("Authorization", "Basic " + base64Credentials);
        return headers;
    }
```

```
    private ResponseEntity<String> launchScheme(String taskDefinition) {
        String requestUri = String.format("http://%1$s:%2$s/tasks/executions", hostName, port);
        HttpHeaders headers = getHttpHeaders();

        HttpEntity<String> requestEntity = new HttpEntity<>(taskDefinition, headers);
        RestTemplate restTemplate = new RestTemplate();
        return restTemplate.exchange(
                requestUri,
                HttpMethod.POST,
                requestEntity,
                String.class);
    }
```

Task definition example:
```
viewing-maker_JULY_1:baseoperation --runner.start-date-to-execute=2018-07-01 --runner.end-date-of-period=2018-07-01 --runner.operation-profile=vm --spring.profiles.active=bo && viewing-filtering_JULY_1:baseoperation --runner.start-date-to-execute=2018-07-01 --runner.end-date-of-period=2018-07-01 --runner.operation-profile=vf --spring.profiles.active=bo
```
SCDF starts composed task runner instead of my "baseoperation" spring boot **task** with next stack trace:

> 2018-09-26 15:58:36.955  INFO 20041 --- [           main] s.c.a.AnnotationConfigApplicationContext : Refreshing org.springframework.context.annotation.AnnotationConfigApplicationContext@17d10166: startup date [Wed Sep 26 15:58:36 MSK 2018]; root of context hierarchy
2018-09-26 15:58:37.379  INFO 20041 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'configurationPropertiesRebinderAutoConfiguration' of type [org.springframework.cloud.autoconfigure.ConfigurationPropertiesRebinderAutoConfiguration$$EnhancerBySpringCGLIB$$c5ba62e2] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2018-09-26 15:58:37.953  INFO 20041 --- [           main] c.c.c.ConfigServicePropertySourceLocator : Fetching config from server at: http://localhost:8888
2018-09-26 15:58:38.101  WARN 20041 --- [           main] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: label not found
2018-09-26 15:58:38.102  INFO 20041 --- [           main] .t.a.c.ComposedtaskrunnerTaskApplication : No active profile set, falling back to default profiles: default
2018-09-26 15:58:38.121  INFO 20041 --- [           main] s.c.a.AnnotationConfigApplicationContext : Refreshing org.springframework.context.annotation.AnnotationConfigApplicationContext@35d176f7: startup date [Wed Sep 26 15:58:38 MSK 2018]; parent: org.springframework.context.annotation.AnnotationConfigApplicationContext@17d10166
2018-09-26 15:58:38.764  INFO 20041 --- [           main] o.s.b.f.s.DefaultListableBeanFactory     : Overriding bean definition for bean 'transactionManager' with a different definition: replacing [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=org.springframework.batch.core.configuration.annotation.SimpleBatchConfiguration; factoryMethodName=transactionManager; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [org/springframework/batch/core/configuration/annotation/SimpleBatchConfiguration.class]] with [Root bean: class [null]; scope=; abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=org.springframework.cloud.task.configuration.SimpleTaskConfiguration; factoryMethodName=transactionManager; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [org/springframework/cloud/task/configuration/SimpleTaskConfiguration.class]]
2018-09-26 15:58:39.174  INFO 20041 --- [           main] o.s.cloud.context.scope.GenericScope     : BeanFactory id=e2b69c27-5d61-3d37-a2fb-93e73489a250
2018-09-26 15:58:39.198  WARN 20041 --- [           main] o.s.c.a.ConfigurationClassEnhancer       : @Bean method ScopeConfiguration.stepScope is non-static and returns an object assignable to Spring's BeanFactoryPostProcessor interface. This will result in a failure to process annotations such as @Autowired, @Resource and @PostConstruct within the method's declaring @Configuration class. Add the 'static' modifier to this method to avoid these container lifecycle issues; see @Bean javadoc for complete details.
2018-09-26 15:58:39.211  WARN 20041 --- [           main] o.s.c.a.ConfigurationClassEnhancer       : @Bean method ScopeConfiguration.jobScope is non-static and returns an object assignable to Spring's BeanFactoryPostProcessor interface. This will result in a failure to process annotations such as @Autowired, @Resource and @PostConstruct within the method's declaring @Configuration class. Add the 'static' modifier to this method to avoid these container lifecycle issues; see @Bean javadoc for complete details.
2018-09-26 15:58:39.398  INFO 20041 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.cloud.autoconfigure.ConfigurationPropertiesRebinderAutoConfiguration' of type [org.springframework.cloud.autoconfigure.ConfigurationPropertiesRebinderAutoConfiguration$$EnhancerBySpringCGLIB$$c5ba62e2] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2018-09-26 15:58:39.413  INFO 20041 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration' of type [org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration$$EnhancerBySpringCGLIB$$a9a05fe5] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2018-09-26 15:58:39.473  INFO 20041 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.cloud.task.batch.configuration.TaskBatchAutoConfiguration' of type [org.springframework.cloud.task.batch.configuration.TaskBatchAutoConfiguration$$EnhancerBySpringCGLIB$$2f100b71] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2018-09-26 15:58:39.486  INFO 20041 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.cloud.task.batch.listener.BatchEventAutoConfiguration' of type [org.springframework.cloud.task.batch.listener.BatchEventAutoConfiguration$$EnhancerBySpringCGLIB$$77661574] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2018-09-26 15:58:40.463  INFO 20041 --- [           main] o.s.jdbc.datasource.init.ScriptUtils     : Executing SQL script from class path resource [org/springframework/cloud/task/schema-postgresql.sql]
2018-09-26 15:58:40.490  INFO 20041 --- [           main] o.s.jdbc.datasource.init.ScriptUtils     : Executed SQL script from class path resource [org/springframework/cloud/task/schema-postgresql.sql] in 26 ms.
2018-09-26 15:58:40.531  INFO 20041 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService  'taskExecutor'
2018-09-26 15:58:40.675  WARN 20041 --- [           main] s.c.a.AnnotationConfigApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'ViewingETL_2018-07-01_2018-07-01_createdAt_2018-09-26_15-58-29-viewing-filtering_JULY_1_0': Unsatisfied dependency expressed through field 'taskOperations'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'taskOperations' defined in class path resource [org/springframework/cloud/task/app/composedtaskrunner/DataFlowConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.cloud.dataflow.rest.client.TaskOperations]: Factory method 'taskOperations' threw exception; nested exception is org.springframework.web.client.HttpClientErrorException: 401 null
2018-09-26 15:58:40.675  INFO 20041 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'taskExecutor'
2018-09-26 15:58:40.708 ERROR 20041 --- [           main] o.s.c.t.listener.TaskLifecycleListener   : An event to end a task has been received for a task that has not yet started.
2018-09-26 15:58:40.736  INFO 20041 --- [           main] utoConfigurationReportLoggingInitializer :
Error starting ApplicationContext. To display the auto-configuration report re-run your application with 'debug' enabled.
2018-09-26 15:58:40.751 ERROR 20041 --- [           main] o.s.boot.SpringApplication               : Application startup failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'ViewingETL_2018-07-01_2018-07-01_createdAt_2018-09-26_15-58-29-viewing-filtering_JULY_1_0': Unsatisfied dependency expressed through field 'taskOperations'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'taskOperations' defined in class path resource [org/springframework/cloud/task/app/composedtaskrunner/DataFlowConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.cloud.dataflow.rest.client.TaskOperations]: Factory method 'taskOperations' threw exception; nested exception is org.springframework.web.client.HttpClientErrorException: 401 null
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:588) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:366) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1264) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:553) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:742) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:867) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:543) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
        at org.springframework.cloud.task.app.composedtaskrunner.ComposedtaskrunnerTaskApplication.main(ComposedtaskrunnerTaskApplication.java:29) [classes!/:1.1.1.RELEASE]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_77]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_77]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_77]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_77]
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [composedtaskrunner-task-1.1.1.RELEASE.jar:1.1.1.RELEASE]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [composedtaskrunner-task-1.1.1.RELEASE.jar:1.1.1.RELEASE]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [composedtaskrunner-task-1.1.1.RELEASE.jar:1.1.1.RELEASE]
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51) [composedtaskrunner-task-1.1.1.RELEASE.jar:1.1.1.RELEASE]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'taskOperations' defined in class path resource [org/springframework/cloud/task/app/composedtaskrunner/DataFlowConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.cloud.dataflow.rest.client.TaskOperations]: Factory method 'taskOperations' threw exception; nested exception is org.springframework.web.client.HttpClientErrorException: 401 null
        at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:599) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1173) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1067) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:513) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:208) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1138) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1066) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:585) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        ... 26 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.cloud.dataflow.rest.client.TaskOperations]: Factory method 'taskOperations' threw exception; nested exception is org.springframework.web.client.HttpClientErrorException: 401 null
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        ... 38 common frames omitted
Caused by: org.springframework.web.client.HttpClientErrorException: 401 null
        at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:78) ~[spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.cloud.dataflow.rest.client.VndErrorResponseErrorHandler.handleError(VndErrorResponseErrorHandler.java:61) ~[spring-cloud-dataflow-rest-client-1.5.0.RELEASE.jar!/:1.5.0.RELEASE]
        at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:700) ~[spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:653) ~[spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:628) ~[spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.web.client.RestTemplate.getForObject(RestTemplate.java:303) ~[spring-web-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.cloud.dataflow.rest.client.DataFlowTemplate.<init>(DataFlowTemplate.java:163) ~[spring-cloud-dataflow-rest-client-1.5.0.RELEASE.jar!/:1.5.0.RELEASE]
        at org.springframework.cloud.dataflow.rest.client.DataFlowTemplate.<init>(DataFlowTemplate.java:145) ~[spring-cloud-dataflow-rest-client-1.5.0.RELEASE.jar!/:1.5.0.RELEASE]
        at org.springframework.cloud.task.app.composedtaskrunner.DataFlowConfiguration.taskOperations(DataFlowConfiguration.java:42) ~[spring-cloud-starter-task-composedtaskrunner-1.1.1.RELEASE.jar!/:1.1.1.RELEASE]
        at org.springframework.cloud.task.app.composedtaskrunner.DataFlowConfiguration$$EnhancerBySpringCGLIB$$550c5e97.CGLIB$taskOperations$0(<generated>) ~[spring-cloud-starter-task-composedtaskrunner-1.1.1.RELEASE.jar!/:1.1.1.RELEASE]
        at org.springframework.cloud.task.app.composedtaskrunner.DataFlowConfiguration$$EnhancerBySpringCGLIB$$550c5e97$$FastClassBySpringCGLIB$$22b8c46d.invoke(<generated>) ~[spring-cloud-starter-task-composedtaskrunner-1.1.1.RELEASE.jar!/:1.1.1.RELEASE]
        at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228) ~[spring-core-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:358) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        at org.springframework.cloud.task.app.composedtaskrunner.DataFlowConfiguration$$EnhancerBySpringCGLIB$$550c5e97.taskOperations(<generated>) ~[spring-cloud-starter-task-composedtaskrunner-1.1.1.RELEASE.jar!/:1.1.1.RELEASE]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_77]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_77]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_77]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_77]
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162) ~[spring-beans-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
        ... 39 common frames omitted

**If  disabling LDAP all is working good**

**Comment by Rostish** [comment]
Collegues?

**Comment by sabbyanandan** [comment]
@Rostish: Thank you for the issue. It is not clear what you're attempting to do, though. Are you using the `DataflowTemplate` and using the `TaskOperations` API directly? Perhaps sharing sample App might be useful.

Here's an [example](https://github.com/spring-cloud/spring-cloud-dataflow-acceptance-tests/blob/master/src/test/java/org/springframework/cloud/dataflow/acceptance/test/AbstractTaskTests.java) of how we are using the operations API directly for the acceptance tests.

**Comment by Rostish** [comment]
@sabbyanandan Good day for you. My app is very simple, i couldn't share it because of NDA.

By steps.
First Variant, LDAP disabled.
1) My app generates task definition.(look at example in the first message)
2) Using RestTemplate(not a DataflowTemplate) i send it to the SCDF rest endpoint to create task.
3) Using RestTemplate(not a DataflowTemplate) i send a command to start this task.
4) My task (baseoperation.jar with needed parameters from definition) starts normally, works normally, ends normally.

Simple Scheme:
String taskDefenition -> Rest request to SCDF to create task -> Rest request to SCDF to start task -> end

Second Variant, LDAP enabled.
1)My app generates task definition.(look at example in the first message)
2.1) I  add to HTTP header basic auth. Needed credentials.
2.2) Using RestTemplate(not a DataflowTemplate) i send it to the SCDF rest endpoint to create task.
2.3) Task definition creates normally. I could see it in dashboard. I could see a graph of this task.

3.1) I  add to HTTP header basic auth. Needed credentials.
3.2) Using RestTemplate(not a DataflowTemplate) i send a command to start this task.
3.3) Task starts, but immediately falls. Composed-task-runner.jar starts intsead of my baseoperation.jar (you can see error message in the first message).


Simple Scheme:
String taskDefenition -> Rest request to SCDF to create task(with credentials) -> Rest request to SCDF to start task(with credentials) -> bad end

**Comment by Rostish** [comment]
Should i  provide additional information?

**Comment by sabbyanandan** [comment]
Hi, @Rostish. My apologies for the delay. We would have to create a sample to replicate this behavior - we will look into it.

**Comment by ghillert** [comment]
Hi @Rostish,

Could you please double-check your setup? I did not see where you provide the security details so that the _Composed Task Runner_ (CTR) can communicate back to the _Spring Cloud Data Flow_ (SCDF) server.

Also, you have to make sure that you are using the latest version of the _Composed Task Runner_  (Version `2.0.0.RELEASE`). Earlier versions did not allow to provide credentials to the CTR

In order to help with your analysis, I have created a quick sample with instructions and an embedded LDAP server to illustrate a complete setup:

https://github.com/ghillert/ldap-composed-task-runner-example

Noting several short-comings, I also created a few issues to improve things in SCDF:

https://github.com/spring-cloud/spring-cloud-dataflow/issues/2503
https://github.com/spring-cloud/spring-cloud-dataflow/issues/2504

**Comment by Rostish** [comment]
Thank you for your example. Unfortunately, it seems to me that we have a misunderstanding.
In my app i WONT to use composed-task-runner. I have my own jar called baseoperation.jar with some buisennes logic. I start it via REST without any security and it works(starts with Spring Boot hello banner).
I start it with enabled LDAP security and it fails(starts with Composed Task Hello Banner).

I use my own restTemplate with headers. I add credential to headers and then use this headers in my requests.

```
private HttpHeaders getHttpHeaders() {
        HttpHeaders headers = new HttpHeaders();
        String plainCredentials = String.format("%1$s:%2$s", userName, password);
        String base64Credentials = Base64.getEncoder().encodeToString(plainCredentials.getBytes());
        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
        headers.add("Authorization", "Basic " + base64Credentials);
        return headers;
    }
```
CreateTask
```
    private JSONObject createNewSCDFTaskPOST(String taskDefinition) throws JSONException {
        HttpHeaders headers = getHttpHeaders(userName, password);
        String requestUri = String.format("http://%1$s:%2$s/tasks/definitions", hostName, port);
        HttpEntity<String> requestEntity = new HttpEntity<>(taskDefinition, headers);

        RestTemplate restTemplate = new RestTemplate();
        ResponseEntity<String> responseEntity = restTemplate.exchange(
                requestUri,
                HttpMethod.POST,
                requestEntity,
                String.class);

        return new JSONObject(responseEntity.getBody());
    }
```
RunTask:
```
    private ResponseEntity<String> launchSchemePOST(String taskDefinition) {
        String requestUri = String.format("http://%1$s:%2$s/tasks/executions", hostName, port);
        HttpHeaders headers = getHttpHeaders(userName, password);

        HttpEntity<String> requestEntity = new HttpEntity<>(taskDefinition, headers);
        RestTemplate restTemplate = new RestTemplate();
        return restTemplate.exchange(
                requestUri,
                HttpMethod.POST,
                requestEntity,
                String.class);
    }
```

In logs i see next:
![image](https://user-images.githubusercontent.com/15521647/47142578-dd6e7d80-d2cb-11e8-855d-76d6419a88ee.png)
![image](https://user-images.githubusercontent.com/15521647/47142604-ef502080-d2cb-11e8-9b43-6285280c4718.png)
![image](https://user-images.githubusercontent.com/15521647/47142636-042cb400-d2cc-11e8-9237-e376556f43e0.png)

**As You see there is Composed Task Hello Banner**
**When i start my own operation there is Spring Boot Hello Banner**

**Comment by sabbyanandan** [comment]
@Rostish: It is easy to go on a different tangent if we aren't able to see your application; hence the reason I had asked for it in the first place - you could still share an anonymized version of the app.

I'm glad you're able to get around the security config problem. (Thanks, @ghillert)

Looking at your definition, though, it appears **you are** using the composed-task functionality. The fact that you have a Task by the name ![image](https://user-images.githubusercontent.com/1562654/47159978-ca47b780-d2a3-11e8-9bb3-064e751440e6.png) and ![image](https://user-images.githubusercontent.com/1562654/47160695-54445000-d2a5-11e8-975e-56283d7da155.png) separated by `&&`, with that, you're instructing SCDF to run these Tasks in sequence one after the other. Please read about the [DSL parser for Composed Task](http://docs.spring.io/spring-cloud-dataflow/docs/1.6.3.RELEASE/reference/htmlsingle/#_composed_tasks_dsl) for more details. 

Of course, when you run the Task-app standalone, it is a plain Boot app - it will run just fine (with Boot banner). When you run it as a composed-task graph in SCDF, the [`composed-task-runner`](https://github.com/spring-cloud-task-app-starters/composed-task-runner) (CTR) will be responsible behind the scenes to run the individual steps in your definition. What you see are the CTR's logs and its banner. Once when the individual steps start to execute, you will see links to log files for each Task apps (likewise, the Boot banner in it).

I'd suggest that you start with a simple graph (eg:  `task create foo --definition "t1: timestamp && t2: timestamp"`). See how that works in your environment. Drill into each of the logs, make sure you're comfortable with it, and then switch to the programmatic approach.

**Comment by Rostish** [comment]
I have a 300+ tasks graphs in production. It works well with spark.
There are no information across all DOC about needed arguments "--dataflow-server-username= --dataflow-server-password=" to run with LDAP. Only few words how to enable LDAP.

**Comment by Rostish** [comment]
In webUI Argument with password is hidden.
![image](https://user-images.githubusercontent.com/15521647/47434979-58440680-d7ac-11e8-88e3-e0fc086271f9.png)

**Comment by sabbyanandan** [comment]
Hello, @Rostish. Apologies for the delay. Quick update. #2516 is addressed in 1.7.x and 2.0.x GA releases (thank you for reporting!). Also, we now have an end-to-end SCDF + CTR + LDAP integration [screencast](https://www.youtube.com/watch?v=dn-1-YzBvMU) and a [sample](https://github.com/spring-cloud/spring-cloud-dataflow-samples/tree/master/security-ldap-uaa-example). 

Check it out and let us know if you have any feedback.
